File: cs/Refactor2/SRP_DI_Workshop/Domain/Revenue/PriceCalculator.cs
namespace SRP_DI_Workshop.Domain.Revenue
{
    public class PriceCalculator : IPriceCalculator
    {
        private readonly IPricingCalculationFactory _pricingCalculationFactory;

        public PriceCalculator(IPricingCalculationFactory pricingCalculationFactory)
        {
            _pricingCalculationFactory = pricingCalculationFactory;
        }

        public void CalculatePrice(OrderItem item, InventoryItem inventoryItem)
        {
            IPricingCalculation pricingCalculation = _pricingCalculationFactory.GetPricingCalculation(inventoryItem.PricingCalculation);
            pricingCalculation.CalculatePrice(item, inventoryItem);
        }
    }
}

File: java/Refactor1/src/main/java/no/bouvet/solid/srpdip/domain/revenue/PriceCalculator.java
package no.bouvet.solid.srpdip.domain.revenue;

import no.bouvet.solid.srpdip.domain.InventoryItem;
import no.bouvet.solid.srpdip.domain.OrderItem;

public class PriceCalculator
{
    public void calculate(OrderItem item, InventoryItem inventoryItem)
    {
        PricingCalculation pricingCalculation;

        switch (inventoryItem.getPricingCalculation())
        {
            case "FULLQUANTITY":
                pricingCalculation = new FullQuantityPricingCalculation();
                break;
            case "BUY2GET1FREE":
                pricingCalculation = new Buy2Get1FreePricingCalculation();
                break;
            default:
                throw new RuntimeException("Invalid pricing calculation type: " + inventoryItem.getPricingCalculation());
        }

        pricingCalculation.calculate(item, inventoryItem);
    }
}
